### PR TITLE
Fix Memory Leak in LogBoxModule

### PR DESCRIPTION
**Pull request by fannnzhang**
## Summary:

To fix [The Memory Leak Issue](https://github.com/facebook/react-native/issues/45080) This change modifies the timing of view creation in the LogModule. The motivation behind this update is to address a potential memory leak issue. Previously, views were being created and held onto, which could lead to references to the Activity being retained even when they were no longer needed. By creating the view only when the show method is called and ensuring it is removed in the hide method, we can prevent these memory leaks and improve the overall memory management and stability of the LogModule. 

Fixes https://github.com/facebook/react-native/issues/45080

- Adjusted the timing of view creation to occur when the `show` method is called.
- Ensured that the created view can be removed in the `hide` method.
- This update addresses potential memory leaks by preventing the view from holding a reference to the Activity.

These changes improve memory management and stability within the LogModule.

Modify the timing of view creation in LogModule. The view is now created when the show method is called, and it can be removed in the hide method. This change resolves potential memory leaks caused by the view holding a reference to the Activity.

## Changelog:

[ANDROID] [FIXED] - Fix LogModule to create view when show is called

### CODE REVIEW

**Comment by dmytrorykun** [comment]
@fannnzhang thank you for the PR! I've red the summary, and the attached issue, but I still need some clarification.
What exactly is leaking in the existing implementation?

**Comment by fannnzhang** [comment]
> @fannnzhang thank you for the PR! I've red the summary, and the attached issue, but I still need some clarification. What exactly is leaking in the existing implementation?

Thank you for your feedback! Let me clarify the existing implementation and the memory leak issue.

In the current implementation, the reference chain is approximately as follows:
```
LogBoxModule -> LogBoxDialogSurfaceDelegate -> mReactRootView -> ReactSurfaceView -> context (current Activity)
```

The issue arises because the current Activity reference is passed to `ReactSurfaceImpl` via the constructor during the creation of the View. However, the reference chain is only broken when the `hide()` method is called in `LogBoxDialogSurfaceDelegate`. Unfortunately, `hide()` is not always guaranteed to be called, which can lead to a memory leak.

When the ReactActivity exits and returns to another native Activity, LeakCanary (the memory leak detection tool) captures this memory leak.

If you need additional details, such as reproduction steps or screenshots from the memory leak detection tool, feel free to ask! 😊

**Review by cortinico — COMMENTED** [review]
Thanks for sending this over @fannnzhang 

Code looks good to me, but I'd like to do more testing around it before we can merge it

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/LogBoxModule.kt [hunk 1]
```diff
@@ -23,20 +23,13 @@ public class LogBoxModule(
       devSupportManager.createSurfaceDelegate(NAME)
           ?: LogBoxDialogSurfaceDelegate(devSupportManager)
 
-  /**
-   * LogBoxModule can be rendered in different surface. By default, it will use LogBoxDialog to wrap
-   * the content of logs. In other platform (for example VR), a surfaceDelegate can be provided so
-   * that the content can be wrapped in custom surface.
-   */
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/LogBoxModule.kt:30]: Can we not lose this comment?

Comment (fannnzhang) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/LogBoxModule.kt:30]: Of course, I've committed the changes to restore the deleted comment. Please review the comment to ensure it's correct.

**Comment by harsh-rksv** [comment]
@fannnzhang  is the issues similar to this? https://github.com/facebook/react-native/issues/45217

**Comment by fannnzhang** [comment]
> @fannnzhang is the issues similar to this? #45217

I think it doesn't seem to be the case. There are quite a few memory leaks in the current new version. So far, I have discovered at least two. One is related to this PR, and the other is in the call chain of FabricUIManager, where a reference to the Activity is held and not released. I will take some time to organize this issue further and might raise an issue and a PR for it later.

**Comment by satheshrgs-rksv** [comment]
@fannnzhang Did you get a chance to look at the memory leak when opening as activity

**Comment by fannnzhang** [comment]
> @fannnzhang Did you get a chance to look at the memory leak when opening as activity

Sure, I will investigate this issue over this weekend. If progress is smooth, I will update you with the relevant details.

**Comment by satheshrgs-rksv** [comment]
@fannnzhang Just checking again if anything to related to this was found ?

**Comment by yzhe554** [comment]
Hey @fannnzhang , any update on the investigation?

**Pull request closed without merging**